docs/aragonOS.md

```diff
@@ -0,0 +1,842 @@
+for better interoperability in the ecosystem.
+
+- **Forwarder:** A Forwarder is a contract that, given some conditions, will pass
+along a certain action to other contract(s).
+Thanks to the fact that proxies allow a certain instance of a contract to never
+have to change its address even if its underlying logic changes, this allows to
+identify a concept such as that an action has been approved in a voting with a
+certain support by a group of holders of a token, just by checking that the
+action **sender address is an instance of a Voting app with a particular address**.
+This helps with the decoupling of authentication and logic explained before.
+
+### 1.2 Architecture: Kernel and apps
+
+An organization or protocol that is built with aragonOS is composed by a set of smart contracts of two types:
+
+- **Kernel:** it is at the core of every organization, there is only one instance of it per organization. It manages one very important mapping to keep track of the different *base contract* address depending on the application, registered apps in the kernel (such as the ACL) or the kernel’s own *base contract*.
+- **Apps:** are contracts that rely on use the kernel for their upgradeability and access control. Apps don’t need to implement any of those as they occur directly in the Kernel or ACL.
+
+### 1.3 Design philosophy
+
+The design philosophy we use when developing Aragon apps is very similar to the UNIX philosophy, we try to architect them to do one thing and one thing well, and to respect and implement the few aragonOS interfaces so that they play nicely with the rest of the ecosystem.
+
+This results in purely technical benefits such as testability, but it is also very powerful when apps are combined and the output of one app becomes the input of an other one (forwarders resemble UNIX pipes in some way).
+
+### 1.4 Lifecycle of an aragonOS call
+
+![](rsc/os3.gif)
+
+## 2. Kernel
+### 2.1 The app mapping
+
+At the core of the kernel lives a mapping, called the `app` mapping, which is very
+critical.
+
+Modifying this mapping can have completely destructive consequences and can result in loss of funds. The permission to execute this action has to be well protected behind the ACL.
+
+```solidity
+function setApp(bytes32 namespace, bytes appId, address app) public;
+```
+
+- **Namespace:** specifies what type of app record is being set.
+- **AppId:** used to identify what app is being set. It is the [ENS `namehash`](http://docs.ens.domains/en/latest/introduction.html#namehash) of the APM repo (e.g. `namehash('voting.aragonpm.eth')`).
+- **App:** Address of a contract that can have different meaning depending on the namespace.
+
+### 2.2 Namespaces
+
+- **Core namespace** (`keccak256('core')`): in this namespace is where the core components of the kernel reside. The only thing in the core mapping is the reference to the kernel base contract.
```

Comment:
change "in this namespace is where the core " to "in this namespace are where the core"